### DISCUSSION TITLE
can't apply this __setattr__ to Proxy object

### DISCUSSION THREAD

**Discussion by endison1986** [kbengine/kbengine#1392] [Q&A]
在Python中定义了一个components如A.py，其中有多个属性，想通过__setattr__监控某些属性的变化，从而计算另一个属性

```python
class A:
    def __setattr__(self, key, value):
        object.__setattr__(self, key, value)
        # 监控变化，修改别的属性，大概意思如下代码
       if key == 'attr1':
              object.__setattr__(self, 'attr3', self.attr1 + self.attr2)
```

这时候就报错了`can't apply this __setattr__ to Proxy object`，请问是我的python代码写的有问题，还是Proxy不支持这种写法，另外还有什么方式可以实现么？

----------------------------------------------------------------------------------------------------------------------------------

现在试了下，发现__setattr__直接进不来了，不管调用self.A=1还是self.setattr('A',1)都不行了，__setattr__就是不进

**Comment by huhubis** [comment]
底层接管了这个方法

**Reply by endison1986** [reply]
奇怪的是之前测试是可以进的，只是会报错，刚又测试了一下直接进不了了。

如果是底层接管的话，那有没有什么好的方式实现，即某个属性变化会重新计算另一个属性，并同步到客户端，因为想把计算方式写在后台，又不想在设置属性的时候每次调用计算的方法

**Reply by kebiao** [reply]
你可以试试在初始化之后去修改self.__setattr__的指向， 记住每次调用之后需要调用引擎之前设置的__setattr__，否则可能破坏底仓

**Reply by endison1986** [reply]
好的，谢谢，我试试

**Comment by qhf566** [comment]
谁有大厂,近5年内开发完的,国内没有运营过的,成品开发完毕的手机游戏,联系我809996677
